File: tulpar-driver/app/src/main/java/kz/tulpartaxi/driver/data/local/db/entity/CityEntity.kt
package kz.tulpartaxi.driver.data.local.db.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "cities")
data class CityEntity(
    @PrimaryKey val id: Int,
    @ColumnInfo(name = "name_kz") val nameKz: String,
    @ColumnInfo(name = "name_ru") val nameRu: String,
    @ColumnInfo(name = "name_en") val nameEn: String,
    val lat: String,
    val lng: String,
    val viewbox: String?,
    val version: Int = 1,
)

File: tulpar-api/src/entities/city.entity.ts
import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity('cities')
export class City {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'name_kz' })
  nameKz: string;

  @Column({ name: 'name_ru' })
  nameRu: string;

  @Column({ name: 'name_en' })
  nameEn: string;

  @Column({ name: 'region_ru', nullable: true })
  regionRu: string;

  @Column({ type: 'decimal', precision: 10, scale: 7 })
  lat: number;

  @Column({ type: 'decimal', precision: 10, scale: 7 })
  lng: number;

  @Column({ nullable: true })
  viewbox: string;

  @Column({ name: 'is_active', default: true })
  isActive: boolean;

  @Column({ default: 1 })
  version: number;
}
